keyboards/bbrfkr/dynamis/keymaps/via/keymap.c

```diff
@@ -20,40 +20,79 @@ enum layer_names {
+      KC_ESC, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, KC_DEL, KC_TRNS,
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+      KC_PSCR, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+      KC_TRNS, KC_LGUI, MO(3),   KC_TRNS,     KC_TRNS,   KC_TRNS, KC_TRNS, KC_RGUI, KC_TRNS,       KC_TRNS, KC_TRNS, KC_MUTE
   ),
 
   [_SETTINGS] = LAYOUT_ansi(
-      RESET, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
-      KC_TRNS, RGB_TOG, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
-      KC_TRNS, RGB_MOD, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
-      KC_TRNS, KC_TRNS, RGB_RMOD, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
-      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS
-  )
+      QK_BOOT, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO,
+      KC_NO, RGB_TOG, RGB_MOD, RGB_RMOD, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO,
+      KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO,
+      KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO,
+      KC_NO, KC_NO, KC_NO,   KC_NO,     KC_NO,   KC_NO, KC_NO, KC_NO, KC_NO,       KC_NO, KC_NO, KC_NO
+  ),
+};
+
+keyevent_t encoder_ccw = {
+    .key = (keypos_t){.row = 10, .col = 1},
+    .pressed = false
 };
+
+keyevent_t encoder_cw = {
+    .key = (keypos_t){.row = 10, .col = 0},
+    .pressed = false
+};
+
+void matrix_scan_user(void) {
+    if (IS_PRESSED(encoder_ccw)) {
+        encoder_ccw.pressed = false;
+        encoder_ccw.time = (timer_read() | 1);
+        action_exec(encoder_ccw);
+    }
+
+    if (IS_PRESSED(encoder_cw)) {
+        encoder_cw.pressed = false;
+        encoder_cw.time = (timer_read() | 1);
+        action_exec(encoder_cw);
+    }
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) { /* First encoder */
+        if (clockwise) {
+            encoder_cw.pressed = true;
+            encoder_cw.time = (timer_read() | 1);
+            action_exec(encoder_cw);
+        } else {
+            encoder_ccw.pressed = true;
+            encoder_ccw.time = (timer_read() | 1);
+            action_exec(encoder_ccw);
+        }
+    }
+    return true;
```

Comment:
Returning false will prevent the kb level code from executing.
```suggestion
    return false;
```

Reply:
Oh, Ok. I fixed it too.